### PR TITLE
[spicedb] Add github action to validate schema

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Adds a github action which validates SpiceDB schema against the defined schema and assertions

## Related Issue(s)

Fixes #

## How to test

Example failure - https://github.com/gitpod-io/gitpod/actions/runs/4044919425/jobs/6955679551
See success in CI run results

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: install/installer/pkg/components/spicedb/data/schema.yaml [hunk 1]
```diff
@@ -32,3 +32,17 @@ schema: |-
   }
 
   definition service {}
+
+# relationships to be used for assertions
+relationships: >-
+  organisation:org1#owner@user:user1
+
+assertions:
```
Comment (easyCZ) [inline install/installer/pkg/components/spicedb/data/schema.yaml:40]: For now, this is pretty limited but this PR focuses on setting it up. We'll iterate on the model and assertions in a follow-up.

File: .github/workflows/permissions.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+name: permissions
+on: pull_request
```
Comment (vulkoingim) [inline .github/workflows/permissions.yml:2]: If you validate only this file - do you think it makes sense to run the action only on when it changes?

```suggestion
on: pull_request
    paths:
      - install/installer/pkg/components/spicedb/data/schema.yaml
```

Comment (easyCZ) [reply .github/workflows/permissions.yml:2]: Sure, makes sense. Is there an easy way to achieve that? Or do I need to dust-off my bash?

Comment (vulkoingim) [reply .github/workflows/permissions.yml:2]: It's the suggestion in my comment 😄 

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

Comment (easyCZ) [reply .github/workflows/permissions.yml:2]: Ah, sorry! I thought it was just how the diff highlighted!

**Review by vulkoingim — APPROVED** [review]

**Review by geropl — APPROVED** [review]
Thanks for adding those tests! :orange_heart:

**Pull request merged**